### ISSUE TITLE
CMake unable to access SDL2 on Ubuntu 20.04

### ISSUE THREAD

**Issue by ElectricTubaphone** [projectM-visualizer/projectm#499]
Hi! I'd like to contribute to the project, but the Cmake build process doesn't complete on my Ubuntu 20.04. When I try to compile the program, CMake reports two errors:
```
CMake Error at CMakeLists.txt:76 (get_target_property):
  get_target_property() called with non-existent target "SDL2::SDL2main".
CMake Error at CMakeLists.txt:85 (message):
  SDL2 libraries were found, but have version .  At least version 2.0.5 is
  required.
```
This is despite the fact that libsdl2-2.0 and libsdl2-dev are installed locally. 'sdl2-config --version' returns 2.0.10. 

I thought the second error might be due to an incorrect comparison between .1 and .5, rather than v10 vs. v5, but the fact that the error report states that the sdl2 version is "." suggests that CMake can't obtain the version.

Furthermore, the CMakeError.log logs as a linker error an "undefined reference to pthread_create", pthread_detach and pthread_join. libpthread-stubs0-dev, the pthread package for apt, is already installed.

Is this something the community would consider an issue proper? If not, I'm sorry to impose.
Thanks!

**Comment by kblaschke** [comment]
Problem lies within SDL2's CMake files, up to 2.0.10 the package config scripts do not add proper targets.

This will be fixed with PR#503 which I just opened as the GitHub Linux build VM also has Ubuntu 20.04 and the issue popped up there too.

**Comment by tongzhipeng** [comment]
It seems it's a bug of sdl https://github.com/libsdl-org/SDL/issues/3531,  and it's fixed in the later version of SDL, There is a workaround before ubuntu 20 update libsdl2-dev or projectM handle it.  
`sudo apt  purge libsdl2-dev` # uninstall SDL 2.0.10 first
`wget  https://www.libsdl.org/release/SDL2-2.0.14.tar.gz` # download 2.0.14 release
then build SDL from source

**Comment by kblaschke** [comment]
Exactly. They partially fixed it in v2.0.12, even if SDL2 is still not following CMake best practices using `install(EXPORT ...)` to automatically generate the target definition files.

I've created a workaround in the PR linked above, basically searching the libraries again and creating the targets. It's far from perfect, but since it only affects older SDL2 versions it will do.
